misk/src/main/kotlin/misk/config/MiskConfig.kt

```diff
@@ -0,0 +1,76 @@
+package misk.config
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException
+import misk.environment.Environment
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URL
+
+object MiskConfig {
+  fun <T : Config> load(
```

Comment:
provide an inline reified version of this and prefer that in samples. e.g.

````kotlin
inline fun <reified T: Config> load(appName: String, environment: Environment) = 
   load(T::class.java, appName, environment)
````

Use

````kotlin
val config = MiskConfig.load<TestConfig>("myapp", Environment.DEVELOPMENT)
````